src/Messages/Common/AbstractVoucherRequest.php

```diff
@@ -80,20 +83,17 @@ public function getData()
      */
     public function sendData($data)
     {
-error_log('[Driver] AbstractVoucherRequest::sendData()');
-error_log('[Driver] The data is: '.var_export($data, true));
+        $this->getGateway()->log('AbstractVoucherRequest::sendData.', ["props"=>$data]);
         $virtualApi = $this->getGateway()->getVirtualApi();
         if ($virtualApi) {
             $response = json_decode($virtualApi->sendRequest($data), true);
-error_log('[Driver] $response is: '.var_export($response, true));
-error_log('[Driver] Got the response; sending it to the listeners');
+            $this->getGateway()->log('$response from virtualapi decoded', ["response"=>var_export($response, true)]);
             // Send all the information to any listeners.
             foreach ($this->getGateway()->getListeners() as $listener) {
-error_log('[Driver] Next listener');
                 $listener->update($this->getListenerAction(), $response);
             }
-
         } else {
+            $this->getGateway()->log('Missing $virtualApi');
```

Comment:
@jamesgarton Maybe we should not mention php variable (with $), but the virtulApi config entry ?

I presume we expect the following config

```
{
	virtualApi: "\DigiTickets\GiftVouchers\Redemption\VirtualApi"
}
```
Is that correct?